Redirect url

kyeumd opened the pull request:
생성된 short url 중 요청이 들어 온 경우
URL 을 검색하여 origin Url로 반환 후 해당 url로 redirect 하는 기능 구현

kyeumd committed:
feat : Redirect URL

new Controller /{shortUrl}

kyeumd committed:
feat : Redirect URL

new Processor, Data, Domain

kyeumd committed:
feat : Redirect URL

new Repository findUrlByOrigin

kyeumd committed:
feat : Redirect URL

반환 객체 MappingUrlData 생성

redirect 기능 구현

f-lab-michael commented on src/main/java/com/url/shorturl/infrastructure/db/ShortUrlRepository.java:
```diff
@@ -1,20 +1,34 @@
 @Component
 public class ShortUrlRepository {
-    private Map<Long, ShortUrl> mappingShortUrl = new HashMap<>();
+    private static Map<Long, ShortUrl> mappingShortUrl = new HashMap<>();
```
멀티 쓰레드 환경에서 사용되는 경우 문제가 될것 같습니다.

kyeumd replied:
88992a0fafbcd85bbb0ed365517d1248d506cb6b
ConcurrentMap , ConcurrentHashMap 으로 변경하였습니다!

f-lab-michael commented on src/main/java/com/url/shorturl/infrastructure/db/ShortUrlRepository.java:
```diff
@@ -1,20 +1,34 @@
+
         mappingUrl.setId(++sequence);
         mappingShortUrl.put(sequence, mappingUrl);
+        log.info("Repo save");
+        log.info("mapping Url info :  " + mappingUrl.toString());
```
두줄로 나눠서 표현하신 이유가 있을까요?

kyeumd replied:
f279ca0c9344d647bb753a9089b5459208859e04
이유가 없어서 로그들 전부 하나로 합쳤습니다!

f-lab-michael commented on src/main/java/com/url/shorturl/presentation/ShortURLController.java:
```diff
@@ -1,23 +1,24 @@
+    private final FindUrlProcessor findUrlProcessor;
 
     @ResponseBody
     @PostMapping("/make/shorturl")
```
make 와 같은 동사를 url 에 사용하는것은 좋은 방법이 아닙니다. 
Post 라고 하는 Http method 가 이미 생성이라는 의미를 가지기 때문에 make 는 생략해도 될것 같아요

kyeumd replied:
48cb2aa06c24b00a51cefc5273ff3e774105f80c
ShortUrlProcessor 로 변경하였습니다!

f-lab-michael replied:
이거 먼가 잘못 전달된것 같은데 url path 에 make 를 제거 하는게 좋겠다는 의견이였습니다.

f-lab-michael commented on src/main/java/com/url/shorturl/presentation/ShortURLController.java:
```diff
@@ -29,4 +30,14 @@ public HttpEntity<MakeShortUrlResponse> makeShortUrl(@RequestBody MakeShortUrlRe
+
+    @GetMapping("/{shortUrl}")
+    public String redirectUrl(@PathVariable String shortUrl) {
+        log.info("redirect url");
+        log.info(shortUrl);
```
이경우도 한줄의 로그로 표현하는게 좋을것 같아요

kyeumd replied:
넵 한줄로 표현했습니다!

f-lab-michael commented on src/main/java/com/url/shorturl/infrastructure/db/ShortUrlRepository.java:
```diff
@@ -1,20 +1,34 @@
         return mappingUrl;
     }
 
+    public static Optional<String> findUrlByOrigin(String shortUrl) {
```
메소드명이 findUrlByShortUrl 이 되야 하는거 아닌가요?

kyeumd replied:
6984a5895bd9dd11bed4b863911b7b2e186e295d
맞습니다 착오가 있었네요..! shortUrl로 수정했습니다!

f-lab-michael commented on src/main/java/com/url/shorturl/domain/UrlFinder.java:
```diff
@@ -0,0 +1,12 @@
+
+public class UrlFinder {
+    public Optional<String> findUrl(String shortUrl) {
+        Optional<String> url = ShortUrlRepository.findUrlByOrigin(shortUrl);
+        return url;
```
이런경우 별도 변수 지정없이 바로 리턴하는걸 추천드립니다.

kyeumd replied:
9f5286abe2cd6787843fe7abddf24fdfc31431c6
inline 으로 변경하여 return 하도록 변경했습니다!

Commit:
refactor : multi thread problem resolve

Change mappingShortUrl

Map -> ConcurrentMap
implements HashMap -> ConcurrentHashMap

Commit:
refactor : logging line refactor

Change 2 line log -> 1 line log

Commit:
refactor : Rename Class

MakeShortUrl to ShortUrlProcessor

Commit:
refactor : Rename Method

findUrlByOrigin to findUrlByShortUrl

Commit:
refactor : inline return

return inline value

Commit:
fix : delete ShortUrl column

ShortUrl 객체의 shortUrl 정보(컬럼) 삭제

Commit:
Revert "fix : delete ShortUrl column"

This reverts commit 62b936ebc0d753c5be3d9526badbd81f7520f467.

Commit:
feat : base62EncodingUrl

base62 방식으로 DB id를 encoding 하여 단축 url 생성

Commit:
feat : update ShortUrl to DB

base62 방식으로 생성된 shortUrl을 DB에 저장

Commit:
refactor : UrlGenerator to UrlEncoder

Generator -> Encoder 클래스 명 수정

Commit:
refactor : Const Base62 Array

base62 인코딩/디코딩에 사용되는 배열 const 로 분리

Commit:
feat : Decoder

base62 디코딩 기능 구현

Commit:
feat : findById

Id로 검색 기능 추가

Commit:
feat : redirectUrl

Short Url을 통한 origin Url 반환

Commit:
feat : redirectUrl

# REST하도록 mapping url 변경

kyeumd committed:
Merge pull request #3 from kyeumd/base62EncodingUrl

Base62 encoding url
